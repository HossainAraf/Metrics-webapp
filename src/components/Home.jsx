// IMPORTS
import React, { useState } from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import asiaImage from '../assets/asia.png';
import CountryFilter from './CountryFilter';
import Card from './Card';
import '../styles/home.css';

// FUNCTION TO CHUNK ARRAY INTO ROWS
function chunkArray(array, chunkSize) {
  const result = [];
  for (let i = 0; i < array.length; i += chunkSize) {
    result.push(array.slice(i, i + chunkSize));
  }
  return result;
}

const Home = () => {
  const [searchTerm, setSearchTerm] = useState('');

  const filteredCountries = CountryFilter(searchTerm);

  return (
    <Container>
      <Row>
        <Col>
          <img src={asiaImage} alt="asia" style={{ maxWidth: '100%' }} />
        </Col>
        <Col id="written-content">
          <h2>ASIA</h2>
          <h5>Population: 4,764,955,914</h5>
        </Col>
      </Row>
      <Row className="center">STATS BY COUNTRY</Row>

      {chunkArray(filteredCountries, 2).map((rowCountries, rowIndex) => {
        const isEvenRow = rowIndex % 2 === 0;

        return (
          <Row key={rowCountries[0].alpha2code}>
            {rowCountries.map((country, colIndex) => {
              const isFirstCol = colIndex === 0;
              let backgroundColor;

              if (isFirstCol) {
                backgroundColor = isEvenRow ? '#dc4681' : '#d04379';
              } else {
                backgroundColor = isEvenRow ? '#d04379' : '#dc4681';
              }

              return (
                <Col className="card-container" key={country.alpha2code} xs={6} md={6} lg={6}>
                  <Card
                    to={`/${country.name}`}
                    title={country.name}
                    backgroundColor={backgroundColor}
                    population={country.population}
                    imageSrc={country.flag}
                  />
                </Col>
              );
            })}
          </Row>
        );
      })}
    </Container>
  );
};

export default Home;
