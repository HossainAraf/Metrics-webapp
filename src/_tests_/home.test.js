import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Home from '../components/Home';

describe('Home Component Snapshot', () => {
  it('should match the snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
