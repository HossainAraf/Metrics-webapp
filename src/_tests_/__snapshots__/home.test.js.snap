// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Snapshot should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <img
        alt="asia"
        src="asia.png"
        style={
          Object {
            "maxWidth": "100%",
          }
        }
      />
    </div>
    <div
      className="col"
      id="written-content"
    >
      <h2>
        ASIA
      </h2>
      <h5>
        Population: 4,764,955,914
      </h5>
    </div>
  </div>
  <div
    className="center row"
  >
    STATS BY COUNTRY
  </div>
</div>
`;
